brian2/tests/test_GSL.py

```diff
@@ -358,7 +358,7 @@ def test_GSL_non_autonomous():
 @with_setup(teardown=reinit_devices)
 @skip_if_not_implemented
 def test_GSL_refractory():
-    eqs = '''dv/dt = 100*Hz : 1 (unless refractory)'''
+    eqs = '''dv/dt = 99.99*Hz : 1 (unless refractory)'''
```

Comment:
What's going on here?

Reply:
We were sloppy about this before, I always slightly tweaked the tests so that they passed. Now that we have made sure that refractoriness is exact up to a time step, this seemed to be an odd thing to do (even more than before, I mean). The problem is that with `dv/dt = 100*Hz` we are supposed to arrive at exactly 1 after 100 time steps, but due to floating point issues, we will end up slightly above or below. E.g. with 100Hz it will be at 1.0000000000000007 after 10ms, but with 1000Hz it will be at 0.9999999999999999 after 1ms... In principle, we could apply a similar kind of shift approach to threshold checks, but this would become messy since the user already provides a string condition. Also, I think that differential equations should rather be used for "realistic" dynamics where you'll never have *exactly* 100Hz as a firing rate. For regular spike trains, rather use `SpikeGeneratorGroup(1, [0], [0]*ms, period=10*ms).`

Reply:
Ah I see, OK that's fine then. In that case, perhaps just a slight performance tweak for numpy and we're done I think.